server/utils.go

```diff
@@ -300,3 +302,26 @@ func getSourceMount(source string, mountinfos []*mount.Info) (path, optional str
 func isContextError(err error) bool {
 	return err == context.Canceled || err == context.DeadlineExceeded
 }
+
+func (s *Server) getResourceOrWait(ctx context.Context, name, resourceType string) (string, error) {
+	if cachedID := s.ResourceFromCache(name); cachedID != "" {
+		log.Infof(ctx, "Found %s %s with ID %s in resource cache; using it", resourceType, name, cachedID)
+		return cachedID, nil
+	}
+	watcher := s.WatcherForResource(name)
+	if watcher == nil {
+		return "", errors.Errorf("error attempting to watch for %s %s: no longer found", resourceType, name)
+	}
+	log.Infof(ctx, "Could not find %s %s in cache; creation not yet finished", resourceType, name)
```

Comment:
Modify the message to indicate that we are waiting for the creation to finish up to timeout value.